collect_app/src/main/java/org/odk/collect/android/mainmenu/MainMenuFragment.kt

```diff
@@ -0,0 +1,311 @@
+import androidx.lifecycle.ViewModelProvider
+import org.odk.collect.android.activities.DeleteSavedFormActivity
+import org.odk.collect.android.activities.FormDownloadListActivity
+import org.odk.collect.android.activities.InstanceChooserList
+import org.odk.collect.android.activities.InstanceUploaderListActivity
+import org.odk.collect.android.activities.WebViewActivity
+import org.odk.collect.android.application.MapboxClassInstanceCreator
+import org.odk.collect.android.databinding.MainMenuBinding
+import org.odk.collect.android.formlists.blankformlist.BlankFormListActivity
+import org.odk.collect.android.formmanagement.FormFillingIntentFactory
+import org.odk.collect.android.gdrive.GoogleDriveActivity
+import org.odk.collect.android.projects.ProjectIconView
+import org.odk.collect.android.projects.ProjectSettingsDialog
+import org.odk.collect.android.utilities.ApplicationConstants
+import org.odk.collect.android.utilities.PlayServicesChecker
+import org.odk.collect.androidshared.data.getState
+import org.odk.collect.androidshared.ui.DialogFragmentUtils
+import org.odk.collect.androidshared.ui.SnackbarUtils
+import org.odk.collect.androidshared.ui.multiclicksafe.MultiClickGuard
+import org.odk.collect.projects.Project
+import org.odk.collect.settings.SettingsProvider
+import org.odk.collect.settings.keys.ProjectKeys
+import org.odk.collect.strings.R
+
+class MainMenuFragment(
+    private val viewModelFactory: ViewModelProvider.Factory,
+    private val settingsProvider: SettingsProvider
+) : Fragment() {
+
+    private lateinit var mainMenuViewModel: MainMenuViewModel
+    private lateinit var currentProjectViewModel: CurrentProjectViewModel
+    private lateinit var permissionsViewModel: RequestPermissionsViewModel
+
+    private val formEntryFlowLauncher =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+            displayFormSavedSnackbar(it.data?.data)
+        }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        val viewModelProvider = ViewModelProvider(this, viewModelFactory)
+        mainMenuViewModel = viewModelProvider[MainMenuViewModel::class.java]
+        currentProjectViewModel = viewModelProvider[CurrentProjectViewModel::class.java]
+        permissionsViewModel = viewModelProvider[RequestPermissionsViewModel::class.java]
+
+        setHasOptionsMenu(true)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return MainMenuBinding.inflate(inflater, container, false).root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        currentProjectViewModel.currentProject.observe(viewLifecycleOwner) { (_, name): Project.Saved ->
+            requireActivity().invalidateOptionsMenu()
+            requireActivity().title = name
+        }
+
+        val binding = MainMenuBinding.bind(view)
```

Comment:
This is just a local variable though so the reference will be freed up (allowing the binding to be GC'd) once the method (`onViewCreated`) has run. I'm just referencing the binding locally (`onResume` uses `MainMenuBinding.bind(requireView())` for instance) rather than keeping a field to avoid having to do anything in `onDestroyView`.